cmd/really/really.cpp

```diff
@@ -137,9 +138,16 @@ class ReallyServer : public quicr::ServerDelegate
 
     qtransport::TransportConfig tcfg{ .tls_cert_filename = "./server-cert.pem",
                                       .tls_key_filename = "./server-key.pem" };
-    server =
-      std::make_unique<quicr::QuicRServer>(relayInfo, tcfg, *this, logger);
+    server = std::make_unique<quicr::Server>(
+      relayInfo, tcfg, shared_from_this(), logger);
+  }
+
+public:
+  static std::shared_ptr<ReallyServer> create()
```

Comment:
Couldn't the caller of this just `make_shared` themselves?

Reply:
They could, but follow best practices as per https://en.cppreference.com/w/cpp/memory/enable_shared_from_this

If you use enable_shared_from_this then it's best to not make it constructible to avoid people trying to create non-shared-ptr objects

Reply:
Okay cool, I figured I was missing something.

Reply:
Yep, that shared_from_this code was broken, so I removed it, since it would be impossible with that circular dependency setup.

That's why the best practice when using shared_from_this is to make the constructor private, and have a static creation method so that the only way to create the object is as a shared_ptr.